src/services/filters.ts

```diff
@@ -55,3 +63,146 @@ export function isFilterMetadata(filter: { value: string; valueLabels?: string[]
     : filter.value;
   return value === filter.valueLabels?.[0];
 }
+
+/**
+ * Filters json parser prop filters that are not currently used in either the line filters or the field filters
+ * @todo unit test
+ */
+export function filterUnusedJSONParserProps(sceneRef: SceneObject) {
+  const lineFormatVar = getLineFormatVariable(sceneRef);
+  const lineFormatFilters = lineFormatVar.state.filters;
+  const fieldsVar = getFieldsVariable(sceneRef);
+  const jsonVariable = getJsonFieldsVariable(sceneRef);
+  const lineFormatSet = new Set();
+  lineFormatFilters.forEach((lineFormatFilter) => {
+    lineFormatSet.add(lineFormatFilter.key);
+  });
+
+  const fieldsFilterSet = new Set();
+  fieldsVar.state.filters.forEach((fieldFilter) => fieldsFilterSet.add(fieldFilter.key));
+
+  // Loop through the json variable filters, remove them if there aren't any fields filters or line format filters that match
+  const filters = jsonVariable.state.filters.filter((jsonParserPropFilter) => {
+    let hasLineFormat = false;
+    let lineFormatKeys: string[] = [];
+
+    // Concat the line format variables together and check to see if the json parser filter match
+    hasLineFormat = lineFormatFilters.some((lineFormatFilter) => {
+      lineFormatKeys.push(lineFormatFilter.key);
+      const concatLineFormatKeys = lineFormatKeys.join('_');
+      return concatLineFormatKeys === jsonParserPropFilter.key;
+    });
+
+    // And check if the fields contain a matching key
+    const hasFieldFilter = fieldsFilterSet.has(jsonParserPropFilter.key);
+    return hasFieldFilter || hasLineFormat;
+  });
+
+  jsonVariable.setState({
+    filters: filters,
+  });
+}
+
+export function removeJsonDrilldownFilters(sceneRef: SceneObject) {
+  const lineFormatVar = getLineFormatVariable(sceneRef);
+  filterUnusedJSONParserProps(sceneRef);
+  lineFormatVar.setState({
+    filters: [],
+  });
+}
+
+export function formatJsonKey(key: string) {
+  if (key.match(LABEL_NAME_INVALID_CHARS)) {
+    key = key.replace(LABEL_NAME_INVALID_CHARS, '_');
+  }
+  return key;
+}
+
+export function addJsonParserFieldValue(sceneRef: SceneObject, keyPath: KeyPath) {
+  const jsonVariable = getJsonFieldsVariable(sceneRef);
+
+  const value = getJsonKeyPath(keyPath);
+  const key = getJsonKey(keyPath);
+  const filterKey = formatJsonKey(key);
+  const nextKeyPath = [...keyPath];
+  let nextKey = nextKeyPath.shift();
+
+  let filters = [
```

Comment:
For example, here we're calling them what they are, filter, not props or prop filters.